authors/apps/articles/views.py

```diff
@@ -57,3 +61,72 @@ def destroy(self, request, pk, **kwargs):
                 return Response("You have succesfully deleted the article")
             return Response("You do not have permision to delete the article")
         return Response(self.err_message, status=status.HTTP_404_NOT_FOUND)
+
+
+class Like(APIView):
+    permission_classes = (IsAuthenticated, )
+
+    @classmethod
+    def post(self, request, **kwargs):
+        current_user = request.user.username
+        article = Article.objects.filter(id=kwargs['article_id'])
+        if article:
+            my_like_already = Likes.objects.filter(
+                article_id=kwargs['article_id'], reviewer_id=current_user, like_or_dislike=1)
+            if not my_like_already:
+                data = {}
+                data['reviewer'] = current_user
+                data['like_or_dislike'] = 1
+                data['article'] = article.first().id
+                serializer = LikeSerializer(data=data)
+                serializer.is_valid(raise_exception=True)
+                serializer.save()
+                Article.objects.filter(
+                    id=kwargs['article_id']).update(likes_count=+1)
+                return Response(
+                    {
+                        "success": "You have successfully liked this article.",
+                        "like": serializer.data
+                    }
+                )
+            return Response(
+                {
+                    "error": "Like rejected",
+                    "message": "You already liked this article",
+                }, status=status.HTTP_400_BAD_REQUEST,)
```

Comment:
@KabohaJeanMark I don't think this is the right status code to return. A `400` is a bad request. Which would mainly mean the `request` or `json` is poorly formatted and cannot be executed by the system which is not the case here.